src/modules/html/producer/html_producer.cpp

```diff
@@ -69,27 +71,116 @@
 
 namespace caspar { namespace html {
 
+struct presentation_frame
+{
+    core::mutable_frame frame;
+    int64_t             audio_pts;
+
+    presentation_frame()
+        : frame(core::mutable_frame(nullptr,
+                                    std::vector<caspar::array<uint8_t>>(),
+                                    caspar::array<int32_t>(),
+                                    core::pixel_format_desc()))
+        , audio_pts(0)
+    {
+    }
+
+    presentation_frame(core::mutable_frame&& frame)
+        : frame(std::move(frame))
+        , audio_pts(0)
+    {
+    }
+
+    presentation_frame(presentation_frame&& other)
+        : frame(std::move(other.frame))
+        , audio_pts(other.audio_pts)
+    {
+    }
+
+    presentation_frame(const presentation_frame&)            = delete;
+    presentation_frame& operator=(const presentation_frame&) = delete;
+
+    presentation_frame& operator=(presentation_frame&& rhs)
+    {
+        frame     = std::move(rhs.frame);
+        audio_pts = rhs.audio_pts;
+        return *this;
+    }
+
+    ~presentation_frame() {}
+
+    void set_audio(caspar::array<int32_t>&& data, int64_t pts)
+    {
+        frame.set_audio_data(std::move(data));
```

Comment:
```suggestion
        frame.audio_data() = std::move(data);
```

Reply:
Obviously! Good catch 👍  I'm still not 100% used to the rvalue special functions. That just shows how long it's been since i actually coded c++ on a regular basis before this.